modules/gltf/gltf_document.cpp

```diff
@@ -6488,55 +6622,77 @@ GLTFAnimation::Track GLTFDocument::_convert_animation_track(Ref<GLTFState> p_sta
 			p_track.scale_track.values.resize(key_count);
 			p_track.scale_track.interpolation = gltf_interpolation;
 
-			for (int32_t key_i = 0; key_i < key_count; key_i++) {
-				Vector3 scale_track = p_animation->track_get_key_value(p_track_i, key_i);
-				p_track.scale_track.values.write[key_i] = scale_track;
+			if (gltf_interpolation == GLTFAnimation::INTERP_CUBIC_SPLINE) {
+				gltf_interpolation = GLTFAnimation::INTERP_LINEAR;
+				p_track.scale_track.times.clear();
+				p_track.scale_track.values.clear();
+				// CATMULLROMSPLINE or CUBIC_SPLINE have to be baked, apologies.
+				const double increment = 1.0 / BAKE_FPS;
+				double time = 0.0;
+				bool last = false;
+				while (true) {
+					Vector3 scale;
+					Error err = p_animation->scale_track_interpolate(p_track_i, time, &scale);
+					ERR_CONTINUE(err != OK);
+					p_track.scale_track.values.push_back(scale);
+					p_track.scale_track.times.push_back(time);
+					if (last) {
+						break;
+					}
+					time += increment;
+					if (time >= anim_end) {
+						last = true;
+						time = anim_end;
+					}
+				}
+			} else {
+				for (int32_t key_i = 0; key_i < key_count; key_i++) {
+					Vector3 scale_track = p_animation->track_get_key_value(p_track_i, key_i);
+					p_track.scale_track.values.write[key_i] = scale_track;
+				}
 			}
 		}
 	} else if (track_type == Animation::TYPE_BEZIER) {
+		const int32_t keys = p_animation->track_get_key_time(p_track_i, key_count - 1) * BAKE_FPS;
```

Comment:
this will bake the bezier with the length of the first encountered track (in the case that :x, :y and :z tracks have different lengths.

bezier is hard enough to work with in Godot that I don't know how worthwhile it is to fix this, but just pointing it out. Users could also be encouraged to use the same number of keys for each of :x, :y and :z

Reply:
I believe the current code doesn't change the algorithm of baking. Need a design clarification at some point.